src/sentry/plugins/providers/integration_repository.py

```diff
@@ -102,13 +102,19 @@ def handle_api_error(self, error):
     def get_config(self, organization):
         raise NotImplementedError
 
-    def validate_config(self, organization, config):
+    def add_repository_data(self, organization, config):
+        """
+        Gets the necessary repository data through the integration's API
+        """
         return config
 
-    def create_repository(self, organization, data):
+    def build_repository_config(self, organization, data):
+        """
+        Builds final dict containing all necessary data to create the repository
+        """
         raise NotImplementedError
 
-    def delete_repository(self, repo):
+    def delete_repository_webhook(self, repo):
```

Comment:
i think it's supposed to be a generic "take any action specific to the integration that should happen when we're deleting the repo"

so imo, we shouldn't rename this as it gets rid of the generality of the method. if we did want to rename it, maybe something like `before_delete_repo` or `on_delete_repo` makes more sense.

Reply:
For context, I suggested changing the name because `delete_repository` didn't seem generic to me – it's describing a very specific action that the function wasn't actually doing. Which was confusing.

I like the `on_delete_repo` idea a lot, though.